app/lib/account/backend.dart

```diff
@@ -682,3 +659,63 @@ Future<void> purgeAccountCache({
     cache.publisherPage(userId).purgeAndRepeat(),
   ]);
 }
+
+/// Low-level, narrowly typed data access methods for [UserSession] entity.
+extension UserSessionDatastoreDBExt on DatastoreDB {
+  _UserSessionDataAccess get userSessions => _UserSessionDataAccess(this);
+}
+
+extension UserSessionTransactionWrapperExt on TransactionWrapper {
+  _UserSessionTransactionDataAcccess get userSessions =>
+      _UserSessionTransactionDataAcccess(this);
+}
+
+class _UserSessionDataAccess {
+  final DatastoreDB _db;
+
+  _UserSessionDataAccess(this._db);
+
+  Future<UserSession?> lookupOrNull(String sessionId) async {
+    final key = _db.emptyKey.append(UserSession, id: sessionId);
+    return await _db.lookupOrNull<UserSession>(key);
+  }
+
+  /// Scans Datastore for all sessions the user has, and invalidates
+  /// them all (by deleting the Datastore entry and purging the cache).
+  Future<void> expireAllForUserId(String userId) async {
+    final query = _db.query<UserSession>()..filter('userId =', userId);
+    final sessionsToDelete = await query.run().toList();
+    for (final session in sessionsToDelete) {
+      await expire(session.sessionId);
+    }
+  }
+
+  /// Removes the session data from the Datastore and from cache.
+  Future<void> expire(String sessionId) async {
+    final key = _db.emptyKey.append(UserSession, id: sessionId);
+    try {
+      await _db.commit(deletes: [key]);
+    } catch (_) {
```

Comment:
Can we get away with catching only Exceptions?

Reply:
Updated.